src/Game-core/sound.cpp

```diff
@@ -0,0 +1,79 @@
+#include "sound.h"
+#include <iostream>
+
+
+
+sound::sound()
+{
+	isDeath = false;
+	isFood = false;
+	if (!eatSoundBuffer.loadFromFile(eatSoundPath))
+	{
+		std::cout << "hit sound file is either corrupted or location given is wrong" << std::endl;
+	}
+
+	if (!deathSoundBuffer.loadFromFile(deathSoundPath))
+	{
+		std::cout << "death sound file is either corrupted or location given is wrong" << std::endl;
+	}
+	
+	if (!bgm.openFromFile(BGMPath))
+	{
+		std::cout << "BGM sound file is either corrupted or location given is wrong" << std::endl;
+	}
+
+	if (!MM.openFromFile(menuMusicPath))
+	{
+		std::cout << "main menu sound file is either corrupted or location given is wrong" << std::endl;
+	}
+
+	eatSound.setBuffer(eatSoundBuffer);	
+	deathSound.setBuffer(deathSoundBuffer);
+	bgm.setVolume(60);
+	MM.setVolume(60);
+}
+
+
+void sound::onCollisionSound()
+{
+	if (isDeath)
+	{
+		deathSound.play();
+		while (deathSound.getStatus() == sf::Sound::Playing)
+		{
+			sf::sleep(sf::milliseconds(100));
+		}
+	}
+	if (isFood)
+	{
+		eatSound.play();
+	}
+	isDeath = false;
+	isFood = false;
+}
+
+void sound::BGM()
+{
+	if(bgm.getStatus() == sf::Music::Playing)
+	{
+		bgm.stop();
+		return;
+	}
+	bgm.play();
+	bgm.setLoop(true);
+}
+
+void sound::menuMusic(bool x)
```

Comment:
Better design is to split this into Start and Stop methods. Or alternatively change the signature to something like this-

```
enum MusicStatus
{
    Start,
    Stop
}
setMenuMusic(MusicStatus status);
```

Reply:
I divided the method into 2 parts i.e start and stop

Reply:
Same thing as before passing the bool, What is it doing?
Else rename the method as SetMenuMusic. That way it becomes
`setMenuMusic(true)` which anyone reading the code will know that it's going to loop the menu music as opposed to `menuMusic(true)` which can be anything like should the music loop? use 5.1 channel? etc